### PR TITLE
Price validation

### PR DESCRIPTION
**Pull request by grandizzy**
Add price validation, some code refactoring
Removed brownie tests and replaced with gas tests
Run brownie gas test on PRs

### CODE REVIEW

File: src/_test/ERC20PoolBid.t.sol [hunk 1]
```diff
@@ -38,26 +38,45 @@ contract ERC20PoolBidTest is DSTestPlus {
...
+            address(lender),
+            3_000 * 1e18,
+            3_010.892022197881557845 * 1e18
+        );
+        lender.addQuoteToken(
+            pool,
+            address(lender),
+            3_000 * 1e18,
+            1_004.989662429170775094 * 1e18
+        );
```
Comment (EdNoepel) [inline src/_test/ERC20PoolBid.t.sol:58]: Is the linter doing this?  It was only 73 columns, and much harder to read when collapsed.

File: src/libraries/BucketMath.sol [hunk 2]
```diff
@@ -89,10 +94,10 @@ library BucketMath {
     /// @dev Price needs to be cast to int, since indices can be negative
     /// @return A boolean indicating if the given price is valid
     function isValidPrice(uint256 _price) public pure returns (bool) {
-        // cast uint256 price to int256 to enable comparison to int constants
-        int256 int_price = int256(_price);
+        int256 index = priceToIndex(_price);
+        uint256 price = indexToPrice(index);
 
-        return (int_price >= MIN_PRICE && int_price < MAX_PRICE);
+        return _price == price;
```
Comment (EdNoepel) [inline src/libraries/BucketMath.sol:100]: This is the important bit; I'd like the tests I'm working on to include this change.

**Review by EdNoepel — APPROVED** [review]
Happy with the parameter name cleanup and additional testing.  One comment regarding overzealous linting.

File: src/_test/ERC20PoolBid.t.sol (hunk 1 above)
Comment (grandizzy) [reply src/_test/ERC20PoolBid.t.sol:58]: yeah, linter is doing, should we change?

**Review by ith-harvey — APPROVED** [review]
LGTM

**Pull request merged**